[WinForms] ListViews with lots of items do not work properly

## Steps to Reproduce

1. Create a Form and add a ListView to it
2. Add a small number of items (less than enough to put an item on the next column), and it works properly
3. Add more items (more than enough to put an item on the next column), and it does not work properly. The items in the last column are shown, and trying to pick an item (that should be) in another column results in a crash.

See [my demo GitHub repo](https://github.com/Talon1024/ListViewBugDemo) for more info.

### Current Behavior

The items in the last column are shown, and items in other columns are not.

### Expected Behavior

The items are broken into columns, and the user can pick any item using their mouse button or keyboard arrow keys. If there are too many items in the list, a scrollbar is shown at the bottom of the ListView in order to allow the user to take a look at items they cannot see.

## On which platforms did you notice this

[ ] macOS (did not test)
[x] Linux
[ ] Windows

**Version Used**: 6.0.0.327

## Stacktrace

This happens when I try to select an item in another column with the arrow keys:

```
Unhandled Exception:
System.ArgumentOutOfRangeException: '-1' is not a valid value for 'Value'. 'Value' should be between 'Minimum' and 'Maximum'
Parameter name: Value
  at System.Windows.Forms.ScrollBar.set_Value (System.Int32 value) [0x00027] in <8678fc7024f54e58bd7ccb95b962b412>:0 
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.ScrollBar.set_Value(int)
  at System.Windows.Forms.ListView.EnsureVisible (System.Int32 index) [0x000d1] in <8678fc7024f54e58bd7ccb95b962b412>:0 
  at System.Windows.Forms.ListView.SelectIndex (System.Int32 display_index) [0x00041] in <8678fc7024f54e58bd7ccb95b962b412>:0 
  at System.Windows.Forms.ListView.HandleNavKeys (System.Windows.Forms.Keys key_data) [0x00085] in <8678fc7024f54e58bd7ccb95b962b412>:0 
  at System.Windows.Forms.ListView.InternalPreProcessMessage (System.Windows.Forms.Message& msg) [0x0001c] in <8678fc7024f54e58bd7ccb95b962b412>:0 
  at System.Windows.Forms.Control.PreProcessMessage (System.Windows.Forms.Message& msg) [0x00000] in <8678fc7024f54e58bd7ccb95b962b412>:0 
  at System.Windows.Forms.Control.PreProcessControlMessageInternal (System.Windows.Forms.Message& msg) [0x0005c] in <8678fc7024f54e58bd7ccb95b962b412>:0 
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.Control.PreProcessControlMessageInternal(System.Windows.Forms.Message&)
  at System.Windows.Forms.Application.RunLoop (System.Boolean Modal, System.Windows.Forms.ApplicationContext context) [0x0022d] in <8678fc7024f54e58bd7ccb95b962b412>:0 
  at System.Windows.Forms.Form.ShowDialog (System.Windows.Forms.IWin32Window owner) [0x001b7] in <8678fc7024f54e58bd7ccb95b962b412>:0 
  at System.Windows.Forms.Form.ShowDialog () [0x00000] in <8678fc7024f54e58bd7ccb95b962b412>:0 
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.Form.ShowDialog()
  at ListViewTest.ListViewTest.Main (System.String[] args) [0x00007] in <07dc989281c14233bfd1fb69af15cf6c>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.ArgumentOutOfRangeException: '-1' is not a valid value for 'Value'. 'Value' should be between 'Minimum' and 'Maximum'
Parameter name: Value
  at System.Windows.Forms.ScrollBar.set_Value (System.Int32 value) [0x00027] in <8678fc7024f54e58bd7ccb95b962b412>:0 
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.ScrollBar.set_Value(int)
  at System.Windows.Forms.ListView.EnsureVisible (System.Int32 index) [0x000d1] in <8678fc7024f54e58bd7ccb95b962b412>:0 
  at System.Windows.Forms.ListView.SelectIndex (System.Int32 display_index) [0x00041] in <8678fc7024f54e58bd7ccb95b962b412>:0 
  at System.Windows.Forms.ListView.HandleNavKeys (System.Windows.Forms.Keys key_data) [0x00085] in <8678fc7024f54e58bd7ccb95b962b412>:0 
  at System.Windows.Forms.ListView.InternalPreProcessMessage (System.Windows.Forms.Message& msg) [0x0001c] in <8678fc7024f54e58bd7ccb95b962b412>:0 
  at System.Windows.Forms.Control.PreProcessMessage (System.Windows.Forms.Message& msg) [0x00000] in <8678fc7024f54e58bd7ccb95b962b412>:0 
  at System.Windows.Forms.Control.PreProcessControlMessageInternal (System.Windows.Forms.Message& msg) [0x0005c] in <8678fc7024f54e58bd7ccb95b962b412>:0 
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.Control.PreProcessControlMessageInternal(System.Windows.Forms.Message&)
  at System.Windows.Forms.Application.RunLoop (System.Boolean Modal, System.Windows.Forms.ApplicationContext context) [0x0022d] in <8678fc7024f54e58bd7ccb95b962b412>:0 
  at System.Windows.Forms.Form.ShowDialog (System.Windows.Forms.IWin32Window owner) [0x001b7] in <8678fc7024f54e58bd7ccb95b962b412>:0 
  at System.Windows.Forms.Form.ShowDialog () [0x00000] in <8678fc7024f54e58bd7ccb95b962b412>:0 
  at (wrapper remoting-invoke-with-check) System.Windows.Forms.Form.ShowDialog()
  at ListViewTest.ListViewTest.Main (System.String[] args) [0x00007] in <07dc989281c14233bfd1fb69af15cf6c>:0
```

--- a/mcs/class/System.Windows.Forms/System.Windows.Forms/ListView.cs
+++ b/mcs/class/System.Windows.Forms/System.Windows.Forms/ListView.cs
@@ -1721,14 +1721,22 @@ namespace System.Windows.Forms
 		void CalculateCustomColumnWidth ()
 		{
 			int min_width = Int32.MaxValue;
+			int col_index_of_min = -1;
 			for (int i = 0; i < columns.Count; i++) {
 				int col_width = columns [i].Width;
 
-				if (col_width < min_width)
+				if (col_width < min_width) {
 					min_width = col_width;
+					col_index_of_min = i;
+				}
 			}
 
-			custom_column_width = min_width;
+			if (min_width >= 0) // manual width
+				custom_column_width = min_width;
+			else if (col_index_of_min != -1) // automatic width, either -1 or -2
+				custom_column_width = GetChildColumnSize(col_index_of_min).Width;
+			else
+				custom_column_width = 0;
 		}
 
 		void LayoutIcons (Size item_size, bool left_aligned, int x_spacing, int y_spacing)

